src/BizHawk.Emulation.Cores/Consoles/Coleco/SN76489col.cs

```diff
@@ -160,14 +160,14 @@ public void generate_sound(int cycles_to_do)
 						if (noise_type)
 						{
 							int bit = (noise & 1) ^ ((noise >> 1) & 1);
-							noise = noise >> 1;
```

Comment:
This if/else should be refactored further:
```
noise = (noise >> 1) | ((noise_type ? (noise & 1) ^ ((noise >> 1) & 1) : noise & 1) << 14);
```
It also kinda looks performance critical since it's doing bit shifting, you could probably optimize it further if you can do an unsafe byte cast on the noise_type because you could then raise noise & 1 to noise_type * true case + (1 - noise_type) * false case and turn the branching into pure math

Reply:
Can we not hyperoptimise cores without profiling please

Reply:
for sure I went a bit overboard on the comments but the initial one liner is probably better readable than the original if split back up over a few lines.